### DISCUSSION TITLE
Organize Skript project to make it bigger!

### DISCUSSION THREAD

**Discussion by Olyno** [SkriptLang/Skript#5554] [General]
### Description

Sometimes I feel like I'm lost in Skript's project, like what it need to work, what is planned for the future, what is not planned etc.... because I'm not a lot active on the repository.

I don't know if I'm alone to feel lost, but anyway, maybe it could be good to have a Milestone or a Project on SkriptLang's repository, or simply pin issues, just to organize more the project and be sure where we go. This Milestone or whatether it is should include all things like skript-parser, aliases, how organize next version of Skript (like what we must to do)....
I find issues really mess for that, that's the reason of my suggestion.

**Comment by bensku** [comment]
Sorry for late response - I've not been very active on this repository either.

You're right that we have a few improvements to do. I don't think we can do milestones, because there can be months-long breaks from active development of Skript. When that happens, we'll just release what we have, every once a while. Github projects might be more useful for grouping issues. For planning... not so much, because we cannot plan much beyond routine maintenance due to limited time the team has.

There has been some talk on long-term plans on Skript, but currently I lack the time to organize, let alone implement them. Even if I had time, implementing a Spigot plugin on top of e.g. @Syst3ms's parser is not something I'm really interested in. To me it seems that no one else has been interested in "leading" that project, either.

As it is, future plans of Skript project are unclear. Substatial improvements require so much work that progress in them is slow and uncertain.

**Comment by APickledWalrus** [comment]
I think that if you're looking for something to help out with, I would go to the issues tab and find an issue and try to help resolve it. The issue log is getting longer, and I see issues going without a response. Cleaning up the issues tab could help with organization a good bit.

With skript-parser, I think if we really do plan to implement it, we'd need to be able to have it at the same state the plugin is at now. However, that means porting over a **lot** of syntax, but, it also gives us the opportunity to really organize the project. To me, the least appealing part of doing it would be having to reconstruct the base parts. I think the best bet might be to make a new branch or repo and open a todo-list issue of what needs done. If people could complete different parts of it, whether that's converting syntax, types, etc, I think we could get it done. If we can get something like that set up, I would be more than happy to help out.

**Comment by bensku** [comment]
Unfortunately, I fear we might lack people who are interested in and experienced enough to *design* and organize development of the new Skript plugin. It is not a small task, and needs to be done carefully - otherwise, we might not end up *improving* anything.

You're also right that many issues currently go without responses. Those issues have not been investigated (let alone fixed) by anyone. Helping us with organization might help a bit, but does not address the core problem: we lack manpower to significant new features and fix bugs in timely fashion. Not the first time this happens, and we got over it before, so I'm not terribly worried. But of course, at the moment, users might have to find workarounds on their own.
